### PR TITLE
feat: implement api-charts

### PR DESCRIPTION
**Pull request by romain-gilliotte**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)
- [ ] Ensure that Types have been updated according to your changes (if needed)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/agent/src/agent/routes/access/api-chart.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  ) {
+    super(services, options);
+
+    this.dataSource = dataSource;
+    this.chartName = chartName;
+  }
+
+  setupRoutes(router: Router): void {
+    // Mount both GET and POST, respectively for smart and api charts.
+    const path = `/_charts/${this.chartName}`;
```
[the remark this answers is not in the document]
Comment (jeffladiray) [reply packages/agent/src/agent/routes/access/api-chart.ts:31]: (It also mismatch with _charts_ mentionned in the docs)

**Review by jeffladiray — COMMENTED** [review]
Two small comments outside the code

1. I can add an API chart with an empty string
2. No example in the _example agent? :(

File: packages/agent/src/agent/routes/access/api-chart.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (romain-gilliotte) [reply packages/agent/src/agent/routes/access/api-chart.ts:31]: > (It also mismatch with _charts_ mentionned in the docs)

good catch

File: docs/frontend-customization/smart-charts/create-a-bar-chart.md [hunk 2]
```diff
@@ -2,81 +2,89 @@
 
 This second example shows how you can achieve any format of charts, as you can benefit from external libraries like D3js.
 
-{% code title="Template tab" %}
+{% tabs %} {% tab title="Agent" %}
 
-```handlebars
-<div class='c-smart-view'>{{this.chart}}</div>
+```javascript
+agent.addChart('alphabetFrequency', async (context, resultBuilder) => {
```
[the remark this answers is not in the document]
Comment (romain-gilliotte) [reply docs/frontend-customization/smart-charts/create-a-bar-chart.md:8]: renamed everything to `resultBuilder` for consistency

File: packages/agent/test/agent/routes/index.test.ts [hunk 3]
```diff
@@ -50,7 +50,7 @@ describe('Route index', () => {
   });
 
   describe('makeRoutes', () => {
-    describe('when a data source without relations', () => {
+    describe('with a data source without relations', () => {
```
[the remark this answers is not in the document]
Comment (romain-gilliotte) [reply packages/agent/test/agent/routes/index.test.ts]: removed it from the diff

**Review by jeffladiray — APPROVED** [review]

**Pull request merged**